brave/brave-core
# Added don't ask checkbox to crash report permission ask dialog

fix https://github.com/brave/brave-browser/issues/18103

User will not see this dialog anymore after checking this checkbox.

<img width="463" alt="Screen Shot 2021-09-15 at 9 58 34 AM" src="https://user-images.githubusercontent.com/6786187/133353376-c2dacc97-9f73-4682-9378-bba7bb1e2183.png">



Resolves 

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
`npm run test brave_browser_tests -- --filter=BraveLocalStatePrefsBrowserTest.DefaultLocalStateTest`

1. Launch browser and disable crash report from settings and re-launch
2. Load brave://inducebrowsercrashforrealz/
3. Re-launch and check crash permission ask dialog is launched
4. Check don't ask again checkbox and click `Block`
5. Re-launch to save preferences
5. Load brave://inducebrowsercrashforrealz/ again
6. Re-launch and check crash permission ask dialog is not launched
